### PR TITLE
add event for cache update

### PR DESCRIPTION
**Pull request by icewind1991**
Allows app to also listen to updates and not just inserts

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: lib/private/Files/Cache/Cache.php [hunk 1]
```diff
@@ -230,7 +231,7 @@ public function getFolderContentsById($fileId) {
 	 */
 	public function put($file, array $data) {
 		if (($id = $this->getId($file)) > -1) {
-			$this->update($id, $data);
+			$this->update($id, $data, $file);
```
Comment (kesselb) [inline lib/private/Files/Cache/Cache.php]: https://github.com/nextcloud/server/blob/d16cfb519e267c64b0ed816901b97f097576821a/lib/private/Files/Cache/Cache.php#L309-L315

:boom:

Comment (MorrisJobke) [reply lib/private/Files/Cache/Cache.php]: @icewind1991 🏓

Comment (icewind1991) [reply lib/private/Files/Cache/Cache.php]: https://github.com/nextcloud/server/pull/15413

**Pull request merged**